### ISSUE TITLE
Binance Futures long order number exception

### ISSUE THREAD

**Issue by shyrik13** [ccxt/ccxt#8115]
**UPDATE**: Binance last reply "the last three characters of this orderid was missed" i.e. as I understood ccxt lost (remove) last three characters of order number.

We trying fetch order by this order number '8389765489684725000' (returned from Binance Futures) and symbol 'ETH/USDT'.
But as result Binance Futures returns exception '-2013 Order does not exist', despite of we have that number in exchange order list.

- OS: Docker (Debian GNU/Linux 9 (stretch))
- Programming Language version: nodeJS 13.12.0
- CCXT version: 1.39.33

```

// code to reproduce issue
async function test() {
    const ccxtExchangeClass = ccxt['binance'];

    let binance = new ccxtExchangeClass({
        apiKey : 'appKey',
        secret : 'secret',
        options : { defaultType: 'future', hedgeMode: false }
    });


    // throws error {"message":"OrderNotFound: binance {\"code\":-2013,\"msg\":\"Order does not exist.\"}
    let result = await binance.fetchOrder('8389765489699654000', 'ETH/USDT', []);
}

test();

```

Result of fetchOrders('ETH/USDT') method.
![image](https://user-images.githubusercontent.com/18314696/102081481-f8d86300-3e18-11eb-99e6-a291b59fe672.png)

Binance response attitude to this issue:

> it now returns longer order id, if the client is built by nodejs, it passes the Number.MAX_SAFE_INTEGER limit. it’s recommend to parse into string from response string.

**Comment by kroitor** [comment]
This is a known issue with ETHUSDT order ids on Binance, we're working on the fix and will upload it very soon.

**Comment by kroitor** [comment]
Fixed in CCXT 1.39.42+ / CCXT Pro 0.4.94+.

**Comment by deman4ik** [comment]
@kroitor Still have same issue with `ccxt@1.39.42`

**Comment by kroitor** [comment]
@deman4ik please, paste your verbose output:

- https://github.com/ccxt/ccxt/wiki/FAQ#what-is-required-to-get-help

```JavaScript
'use strict';
const ccxt = require ('ccxt')

console.log ('CCXT Version:', ccxt.version)
const binance = new ccxt.binance({
    'enableRateLimit': true,
    'apiKey': 'YOUR_API_KEY',
    'secret': 'YOUR_SECRET',
    'options': {
        'defaultType': 'spot', // or margin/future/delivery
    }
})

async function main() {
    await exchange.loadMarkets ()
    exchange.verbose = true
    const orders = exchange.fetchOrders ('ETH/USDT')
}

main()
```

**Comment by deman4ik** [comment]
```js
"use strict";
const ccxt = require("ccxt");

console.log("CCXT Version:", ccxt.version);
const exchange = new ccxt.binance({
  enableRateLimit: true,

  apiKey: "*",
  secret: "*",
  options: { defaultType: "future" }
});

async function main() {
  try {
    await exchange.loadMarkets();
    exchange.verbose = true;
    const orders = await exchange.fetchOrders("ETH/USDT", null, 5);
    console.log(orders);

    const checkedFirst = await exchange.fetchOrder(orders[0].id, "ETH/USDT");
    console.log(checkedFirst);
  } catch (e) {
    console.log(e.constructor.name, e.message);
  }
}

main();
```

**RESPONSE**

```js
CCXT Version: 1.39.42
fetch:
 binance GET https://fapi.binance.com/fapi/v1/allOrders?timestamp=1608217598122&recvWindow=5000&symbol=ETHUSDT&startTime=&limit=5&signature=cb4b4ea258459128808543478907d81100c2c1bf3ce0cad6e93a9daff792bbb2 
Request:
 {
  'X-MBX-APIKEY': '*'
} 
 undefined 

handleRestResponse:
 binance GET https://fapi.binance.com/fapi/v1/allOrders?timestamp=1608217598122&recvWindow=5000&symbol=ETHUSDT&startTime=&limit=5&signature=cb4b4ea258459128808543478907d81100c2c1bf3ce0cad6e93a9daff792bbb2 200 OK 
Response:
 {
  'Access-Control-Allow-Credentials': 'true',
  'Access-Control-Allow-Headers': '*',
  'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
  'Access-Control-Allow-Origin': '*',
  'Cache-Control': 'no-cache, no-store, must-revalidate',
  Connection: 'keep-alive',
  'Content-Encoding': 'gzip',
  'Content-Security-Policy': "default-src 'self'",
  'Content-Type': 'application/json',
  Date: 'Thu, 17 Dec 2020 15:06:38 GMT',
  Expires: '0',
  Pragma: 'no-cache',
  Server: 'Tengine',
  'Strict-Transport-Security': 'max-age=31536000; includeSubdomains',
  'Transfer-Encoding': 'chunked',
  Vary: 'Accept-Encoding',
  Via: '1.1 292bf579bbf52fb581ef04e0d0f088e8.cloudfront.net (CloudFront)',
  'X-Amz-Cf-Id': 'P7EJivTZJ0kshWAj5MFUTvCYQDyqIeP7fRelI5WbL3J-ajMVWsUxqw==',
  'X-Amz-Cf-Pop': 'AMS50-C1',
  'X-Cache': 'Miss from cloudfront',
  'X-Content-Type-Options': 'nosniff',
  'X-Frame-Options': 'SAMEORIGIN',
  'X-Mbx-Used-Weight-1m': '6',
  'X-Response-Time': '5ms',
  'X-Xss-Protection': '1; mode=block'
} 
 [{"orderId":8389765489750645071,"symbol":"ETHUSDT","status":"NEW","clientOrderId":"thJl6exJ9Uyrey7p8kZYe3","price":"605.61","avgPrice":"0.00000","origQty":"0.500","executedQty":"0","cumQuote":"0","timeInForce":"GTC","type":"LIMIT","reduceOnly":false,"closePosition":false,"side":"BUY","positionSide":"BOTH","stopPrice":"0","workingType":"CONTRACT_PRICE","priceProtect":false,"origType":"LIMIT","time":1608141832212,"updateTime":1608141832212},{"orderId":8389765489750645157,"symbol":"ETHUSDT","status":"NEW","clientOrderId":"1QqhdzVq9jWtlqvxUVb6Xp","price":"594.97","avgPrice":"0.00000","origQty":"1","executedQty":"0","cumQuote":"0","timeInForce":"GTC","type":"LIMIT","reduceOnly":false,"closePosition":false,"side":"BUY","positionSide":"BOTH","stopPrice":"0","workingType":"CONTRACT_PRICE","priceProtect":false,"origType":"LIMIT","time":1608141832603,"updateTime":1608141832603},{"orderId":8389765489750646660,"symbol":"ETHUSDT","status":"NEW","clientOrderId":"2Qh1ma7twtzvL41Zz7cocn","price":"605.61","avgPrice":"0.00000","origQty":"0.500","executedQty":"0","cumQuote":"0","timeInForce":"GTC","type":"LIMIT","reduceOnly":false,"closePosition":false,"side":"BUY","positionSide":"BOTH","stopPrice":"0","workingType":"CONTRACT_PRICE","priceProtect":false,"origType":"LIMIT","time":1608141836715,"updateTime":1608141836715},{"orderId":8389765489750647251,"symbol":"ETHUSDT","status":"NEW","clientOrderId":"aEnOgLrVspamJY0z9a5nyU","price":"599.20","avgPrice":"0.00000","origQty":"0.500","executedQty":"0","cumQuote":"0","timeInForce":"GTC","type":"LIMIT","reduceOnly":false,"closePosition":false,"side":"BUY","positionSide":"BOTH","stopPrice":"0","workingType":"CONTRACT_PRICE","priceProtect":false,"origType":"LIMIT","time":1608141838462,"updateTime":1608141838462},{"orderId":8389765489781233915,"symbol":"ETHUSDT","status":"NEW","clientOrderId":"rwzxmqklX62RJLOXjGnGeD","price":"100","avgPrice":"0.00000","origQty":"0.100","executedQty":"0","cumQuote":"0","timeInForce":"GTC","type":"LIMIT","reduceOnly":false,"closePosition":false,"side":"BUY","positionSide":"BOTH","stopPrice":"0","workingType":"CONTRACT_PRICE","priceProtect":false,"origType":"LIMIT","time":1608216483975,"updateTime":1608216483975}] 

[
  {
    info: {
      orderId: '8389765489750645071',
      symbol: 'ETHUSDT',
      status: 'NEW',
      clientOrderId: 'thJl6exJ9Uyrey7p8kZYe3',
      price: '605.61',
      avgPrice: '0.00000',
      origQty: '0.500',
      executedQty: '0',
      cumQuote: '0',
      timeInForce: 'GTC',
      type: 'LIMIT',
      reduceOnly: false,
      closePosition: false,
      side: 'BUY',
      positionSide: 'BOTH',
      stopPrice: '0',
      workingType: 'CONTRACT_PRICE',
      priceProtect: false,
      origType: 'LIMIT',
      time: '1608141832212',
      updateTime: 1608141832212
    },
    id: '8389765489750645071',
    clientOrderId: 'thJl6exJ9Uyrey7p8kZYe3',
    timestamp: 1608141832212,
    datetime: '2020-12-16T18:03:52.212Z',
    lastTradeTimestamp: undefined,
    symbol: 'ETH/USDT',
    type: 'limit',
    timeInForce: 'GTC',
    postOnly: false,
    side: 'buy',
    price: 605.61,
    amount: 0.5,
    cost: 0,
    average: undefined,
    filled: 0,
    remaining: 0.5,
    status: 'open',
    fee: undefined,
    trades: undefined
  },
  {
    info: {
      orderId: '8389765489750645157',
      symbol: 'ETHUSDT',
      status: 'NEW',
      clientOrderId: '1QqhdzVq9jWtlqvxUVb6Xp',
      price: '594.97',
      avgPrice: '0.00000',
      origQty: '1',
      executedQty: '0',
      cumQuote: '0',
      timeInForce: 'GTC',
      type: 'LIMIT',
      reduceOnly: false,
      closePosition: false,
      side: 'BUY',
      positionSide: 'BOTH',
      stopPrice: '0',
      workingType: 'CONTRACT_PRICE',
      priceProtect: false,
      origType: 'LIMIT',
      time: '1608141832603',
      updateTime: 1608141832603
    },
    id: '8389765489750645157',
    clientOrderId: '1QqhdzVq9jWtlqvxUVb6Xp',
    timestamp: 1608141832603,
    datetime: '2020-12-16T18:03:52.603Z',
    lastTradeTimestamp: undefined,
    symbol: 'ETH/USDT',
    type: 'limit',
    timeInForce: 'GTC',
    postOnly: false,
    side: 'buy',
    price: 594.97,
    amount: 1,
    cost: 0,
    average: undefined,
    filled: 0,
    remaining: 1,
    status: 'open',
    fee: undefined,
    trades: undefined
  },
  {
    info: {
      orderId: '8389765489750646660',
      symbol: 'ETHUSDT',
      status: 'NEW',
      clientOrderId: '2Qh1ma7twtzvL41Zz7cocn',
      price: '605.61',
      avgPrice: '0.00000',
      origQty: '0.500',
      executedQty: '0',
      cumQuote: '0',
      timeInForce: 'GTC',
      type: 'LIMIT',
      reduceOnly: false,
      closePosition: false,
      side: 'BUY',
      positionSide: 'BOTH',
      stopPrice: '0',
      workingType: 'CONTRACT_PRICE',
      priceProtect: false,
      origType: 'LIMIT',
      time: '1608141836715',
      updateTime: 1608141836715
    },
    id: '8389765489750646660',
    clientOrderId: '2Qh1ma7twtzvL41Zz7cocn',
    timestamp: 1608141836715,
    datetime: '2020-12-16T18:03:56.715Z',
    lastTradeTimestamp: undefined,
    symbol: 'ETH/USDT',
    type: 'limit',
    timeInForce: 'GTC',
    postOnly: false,
    side: 'buy',
    price: 605.61,
    amount: 0.5,
    cost: 0,
    average: undefined,
    filled: 0,
    remaining: 0.5,
    status: 'open',
    fee: undefined,
    trades: undefined
  },
  {
    info: {
      orderId: '8389765489750647251',
      symbol: 'ETHUSDT',
      status: 'NEW',
      clientOrderId: 'aEnOgLrVspamJY0z9a5nyU',
      price: '599.20',
      avgPrice: '0.00000',
      origQty: '0.500',
      executedQty: '0',
      cumQuote: '0',
      timeInForce: 'GTC',
      type: 'LIMIT',
      reduceOnly: false,
      closePosition: false,
      side: 'BUY',
      positionSide: 'BOTH',
      stopPrice: '0',
      workingType: 'CONTRACT_PRICE',
      priceProtect: false,
      origType: 'LIMIT',
      time: '1608141838462',
      updateTime: 1608141838462
    },
    id: '8389765489750647251',
    clientOrderId: 'aEnOgLrVspamJY0z9a5nyU',
    timestamp: 1608141838462,
    datetime: '2020-12-16T18:03:58.462Z',
    lastTradeTimestamp: undefined,
    symbol: 'ETH/USDT',
    type: 'limit',
    timeInForce: 'GTC',
    postOnly: false,
    side: 'buy',
    price: 599.2,
    amount: 0.5,
    cost: 0,
    average: undefined,
    filled: 0,
    remaining: 0.5,
    status: 'open',
    fee: undefined,
    trades: undefined
  },
  {
    info: {
      orderId: '8389765489781233915',
      symbol: 'ETHUSDT',
      status: 'NEW',
      clientOrderId: 'rwzxmqklX62RJLOXjGnGeD',
      price: '100',
      avgPrice: '0.00000',
      origQty: '0.100',
      executedQty: '0',
      cumQuote: '0',
      timeInForce: 'GTC',
      type: 'LIMIT',
      reduceOnly: false,
      closePosition: false,
      side: 'BUY',
      positionSide: 'BOTH',
      stopPrice: '0',
      workingType: 'CONTRACT_PRICE',
      priceProtect: false,
      origType: 'LIMIT',
      time: '1608216483975',
      updateTime: 1608216483975
    },
    id: '8389765489781233915',
    clientOrderId: 'rwzxmqklX62RJLOXjGnGeD',
    timestamp: 1608216483975,
    datetime: '2020-12-17T14:48:03.975Z',
    lastTradeTimestamp: undefined,
    symbol: 'ETH/USDT',
    type: 'limit',
    timeInForce: 'GTC',
    postOnly: false,
    side: 'buy',
    price: 100,
    amount: 0.1,
    cost: 0,
    average: undefined,
    filled: 0,
    remaining: 0.1,
    status: 'open',
    fee: undefined,
    trades: undefined
  }
]
fetch:
 binance GET https://fapi.binance.com/fapi/v1/order?timestamp=1608217598622&recvWindow=5000&symbol=ETHUSDT&orderId=8389765489750645000&signature=66c6ab5a9363ae0a3501da59bb010750d8cec0f6e6eee9c6a75fb8640d396dce 
Request:
 {
  'X-MBX-APIKEY': '*'
} 
 undefined 

handleRestResponse:
 binance GET https://fapi.binance.com/fapi/v1/order?timestamp=1608217598622&recvWindow=5000&symbol=ETHUSDT&orderId=8389765489750645000&signature=66c6ab5a9363ae0a3501da59bb010750d8cec0f6e6eee9c6a75fb8640d396dce 400 Bad Request 
Response:
 {
  'Access-Control-Allow-Methods': 'GET, POST, PUT, DELETE, OPTIONS',
  'Access-Control-Allow-Origin': '*',
  Connection: 'keep-alive',
  'Content-Length': '44',
  'Content-Type': 'application/json',
  Date: 'Thu, 17 Dec 2020 15:06:38 GMT',
  Server: 'Tengine',
  Via: '1.1 292bf579bbf52fb581ef04e0d0f088e8.cloudfront.net (CloudFront)',
  'X-Amz-Cf-Id': 'F9zj5jCkA0hstoGyqVXN0uVp7ZzUk2KIQ7Wse-3nfOqF0rbZcNVt2w==',
  'X-Amz-Cf-Pop': 'AMS50-C1',
  'X-Cache': 'Error from cloudfront',
  'X-Mbx-Used-Weight-1m': '7',
  'X-Response-Time': '2ms'
} 
 {"code":-2013,"msg":"Order does not exist."} 

OrderNotFound binance {"code":-2013,"msg":"Order does not exist."}
```

**Comment by shyrik13** [comment]
> @kroitor Still have same issue with `ccxt@1.39.42`

@kroitor Actually I avoid this with overwriting ccxt fetchOrder method and changing 
request['orderId'] = parseInt (id);
to 
request['orderId'] = id;

**Comment by kroitor** [comment]
@deman4ik @shyrik13 got it, so it accepts strings, i'll push an edit shortly.

**Comment by kroitor** [comment]
@shyrik13 @deman4ik should be ok in 1.39.43+, the auto build will arrive in ~20 minutes, please, let me know if you still have issues with it after upgrading. Thx for the feedback!

**Comment by deman4ik** [comment]
@kroitor `fetchOrder` works but `cancelOrder` not

**Comment by kroitor** [comment]
@deman4ik ouch, missed that one, thx, will push another small fix in a sec.
